sklearn/compose/_column_transformer.py

```diff
@@ -362,41 +368,31 @@ def set_params(self, **kwargs):
         self._set_params("_transformers", **kwargs)
         return self
 
-    def _iter(self, fitted, replace_strings, column_as_strings):
+    def _iter(self, fitted, column_as_strings, filter_drop, filter_empty_columns):
```

Comment:
Could you use `skip_drop` and `skip_empty_columns? "To filter" is sometimes understood as "to select" instead of "to filter out".

Reply:
I would consider replacing `columns_as_strings` with `columns_as_labels` provided it makes sense throughout the code. It is not just about the name, but about semantics and consequently expected and actual behaviour. The conflation between _integers vs strings_ and _positions vs labels_ is problematic, because labels can be integers. See e.g. #22556

Reply:
done